chore(#6): 배포설정(S3 + GitHub Actions CI/CD)

jngmnj opened the pull request:
<!-- 제목 예시: feat(#123): 회원가입 기능 추가 -->

## 🔧 관련 이슈

- 이 PR은 다음 이슈와 관련 있습니다: #6 

## 🔄 변경 사항

* [x] 기타 (환경설정, 의존성 업데이트 등)

## ✔️ 변경 사항 상세 설명

* 변경된 파일:

  * `.github/workflows/deploy.yml`
  * `src/App.tsx`
  * `src/components/common/Header.tsx`
  * `src/pages/QnaDetailPage.tsx`

* 주요 구현/수정 내용:

  * LoginPage 라우터 추가, QnaListPage 상대경로변경
  * QnaDetailPage Button 컴포넌트 props 변경
    * setIsLoggedIn 미사용으로 빌드에러를 해결하기위해 logout함수 추가, 적용(추후 실제 기능개발 필요)
    * 헤더 기본 흰배경 추가
  * Header 리팩토링 
    * setIsLoggedIn 미사용으로 빌드에러를 해결하기위해 logout함수 추가, 적용(추후 실제 기능개발 필요)
    * 헤더 기본 흰배경 추가
  * github action workflows 작성
    * develop 브랜치에 push되면 워크플로우 시작, AWS S3에 업로드, CloudFront 캐시를 무효화
    * CI/CD 파이프라인에서 `npm install` → `npm ci`로 변경하여 **의존성 일관성 확보**
    * 빌드 시 `NODE_ENV=production` 명시로 **최적화된 번들 생성** 유도
    * GitHub Actions 배포 트리거 브랜치를 `feature/*` → `develop`으로 변경하여 **실제 배포 기준에 맞춤**

## 📸 스크린샷 (UI 변경 시 필수)

* UI 변경 사항 없음

## 📝 문서화

* [ ] 관련 문서가 업데이트되었습니다. (`README.md`, Notion 링크 등)

## 🔍 리뷰어에게 요청 사항 (선택)

* CI/CD 워크플로우 변경 내용 중 문제 없는지 검토 부탁드립니다.
* 특히 `NODE_ENV`, 브랜치 기준 변경이 의도대로 동작하는지 확인해주세요.

## ⚠️ 기타 주의 사항

* 이 PR 머지 이후 develop 브랜치에서 push 시 AWS S3 + CloudFront 배포가 자동으로 트리거됩니다.

jngmnj committed:
🚚 build: github action workflows 작성(#6)

jngmnj committed:
♻️ refactor: QnaDetailPage Button 컴포넌트 props 변경(#6)

jngmnj committed:
♻️ refactor: Header 컴포넌트 리팩토링(#6)

- setIsLoggedIn 미사용으로 빌드에러를 해결하기위해 logout함수 추가, 적용(추후 실제 기능개발 필요)
- 헤더 기본 흰배경 추가

jngmnj committed:
💡 chore: CI/CD 안정화를 위한 npm ci 및 NODE_ENV 설정 추가, develop 브랜치 기준으로 수정(#6)

- package 설치 명령어를 npm install → npm ci로 변경하여 일관된 의존성 확보
- 빌드 시 NODE_ENV=production 명시로 최적화된 번들 생성
- GitHub Actions의 배포 트리거 브랜치를 feature → develop으로 수정

ella6010 reviewed (dismissed):
고생하셨어요~

chen4023 commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,52 @@
+          echo "Listing files in the repository:"
+          ls -la
+
+      - name: install & build
+        run: |
+          echo "Building the project..."
+          NODE_ENV=production npm ci
+          npm run build
```
크으.. 👍🏻

chen4023 commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,52 @@
+name: Deploy to S3 and CloudFront
+
+on:
+  push:
+    branches:
+      - develop
```
push 시점은 이미 코드가 합쳐져 있는 경우이기 때문에 이미 병합된 상태 이후 발견하는 것 보다는, PR 시점에는 린트검사를 실행하고,
push 시점에는 자동 배포가 가능하도록 분리하는 방법도 한 번 고려해보시면 좋을 것 같네요!

chen4023 reviewed (dismissed):
고생하셨습니다 정민님!
역시 킥이 하나씩 있군요👍🏻

jngmnj committed:
🚚 build: PR 대상 develop 브랜치에서 린트 검사 수행하도록 CI 워크플로우 추가(#6)

- pull_request 이벤트를 트리거로 하여 PR 생성 및 업데이트 시 lint 수행
- npm ci 기반 의존성 설치 후 npm run lint 실행
- 병합 전 코드 품질 확인을 위한 자동화된 검사 목적

ella6010 reviewed (approved):
화이팅!!

devjaesung reviewed (approved):
배포 설정 수고하셨습니다!
